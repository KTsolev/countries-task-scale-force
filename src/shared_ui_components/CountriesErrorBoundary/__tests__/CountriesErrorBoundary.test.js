import React, { Component } from 'react';
import renderer from 'react-test-renderer';
import { ErrorBoundary } from '../CountriesErrorBoundary';

class ComponentWithError extends Component {
    render() {
      return (
        <div>
          <input type = {null} value = {null}/>  
        </div>
      );
    }
  }

describe('<ErrorBoundary> window',()=> {
  it('should match the snapshot', () => {
    const tree = renderer.create(<ErrorBoundary>Test</ErrorBoundary>);

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('displays error message on error generated by child', () => {

    const tree = renderer.create(
      <ErrorBoundary hasError errorInfo='Something went wrong' > 
        <ComponentWithError />
      </ErrorBoundary> 
      );
      
    expect(tree.toJSON()).toMatchSnapshot();
  });
});