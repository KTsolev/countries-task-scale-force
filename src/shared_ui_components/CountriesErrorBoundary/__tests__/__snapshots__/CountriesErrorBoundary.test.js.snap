// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorBoundary> window displays error message on error generated by child 1`] = `
<div>
  <input
    type={null}
    value={null}
  />
</div>
`;

exports[`<ErrorBoundary> window should match the snapshot 1`] = `"Test"`;
